### PR TITLE
CY-1207 Add sites commands

### PR DESCRIPTION
**Pull request by inbala**

### CODE REVIEW

File: amqp-postgres/dev-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-git+https://github.com/cloudify-cosmo/cloudify-common@master#egg=cloudify-common==master
+git+https://github.com/cloudify-cosmo/cloudify-common@CY-1207-add-sites-commands#egg=cloudify-common==CY-1207-add-sites-commands
```
Comment (inbala) [inline amqp-postgres/dev-requirements.txt]: Revert before the merge

File: rest-service/dev-requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/cloudify-cosmo/cloudify-common@master#egg=cloudify-common==master
+git+https://github.com/cloudify-cosmo/cloudify-common@CY-1207-add-sites-commands#egg=cloudify-common==CY-1207-add-sites-commands
```
Comment (inbala) [inline rest-service/dev-requirements.txt]: Revert before the merge

Comment (inbala) [inline rest-service/dev-requirements.txt]: Revert it before the merge

File: rest-service/manager_rest/rest/resources_v3_1/sites.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        self._validate_lat_and_long(request_dict)
+        return request_dict
+
+    def _validate_lat_and_long(self, request_dict):
+        latitude = request_dict.get('latitude')
+        longitude = request_dict.get('longitude')
+
+        # Optional params
+        if not latitude and not longitude:
+            return None, None
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/sites.py]: The return is not needed or used here

File: rest-service/manager_rest/rest/resources_v3_1/sites.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    def post(self, name):
+        """
+        Update an existing site
+        """
+        request_dict = self._validate_site_params(name)
+        if request_dict.get('new_name'):
+            validate_inputs({'new_name': request_dict['new_name']})
+        storage_manager = get_storage_manager()
+        site = storage_manager.get(models.Site, name)
+        site.name = request_dict.get('new_name') or site.name
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/sites.py]: u can switch this to .get('prop-name',current-value)

File: rest-service/manager_rest/rest/resources_v3_1/sites.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        """
+        Delete an existing site
+        """
+        storage_manager = get_storage_manager()
+        site = storage_manager.get(models.Site, name)
+        return storage_manager.delete(site, validate_global=True)
+
+    def _validate_site_params(self, name):
+        validate_inputs({'name': name})
+        visibility = get_visibility_parameter(
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/sites.py]: there is handling of this prop twice once here and once in 79 line

File: rest-service/manager_rest/test/endpoints/test_sites.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+                                'test_site')
+
+    def test_delete_site_not_found(self):
+        error_msg = "404: Requested `Site` with ID `my_site` was not found"
+        self.assertRaisesRegexp(CloudifyClientError,
+                                error_msg,
+                                self.client.sites.delete,
+                                'my_site')
+
+    def _put_site(self, name='test_site'):
```
Comment (tyacbovi) [inline rest-service/manager_rest/test/endpoints/test_sites.py]: I will put the help funcs first

**Pull request merged**